Lib/re/__init__.py

```diff
@@ -229,6 +231,18 @@ def purge():
     _cache.clear()
     _compile_repl.cache_clear()
 
+def template(pattern, flags=0):
+    "Compile a template pattern, returning a Pattern object, deprecated"
+    import warnings
```

Comment:
Since many modules import `re` I've decided it's better to lazy-import `warnings` here to avoid polluting the modules space for people who do not use this deprecated function. Let me know if that's considered bad.

Reply:
It is a common technique in the re module. All warnings import the warning module lazily.